### PR TITLE
*: remove evaluate.Eval from ddl

### PR DESCRIPTION
**Pull request by hanfei1991**
abandon evaluate.Eval
@shenli @coocood @zimulala PTAL

### CODE REVIEW

File: plan/expression_rewriter.go [hunk 1]
```diff
@@ -20,6 +20,24 @@ import (
...
+// GetInfoSchema gets infoschema from ctx.
+var GetInfoSchema func(ctx context.Context) infoschema.InfoSchema
+
+// EvalAstExpr evaluates ast expression directly.
+func EvalAstExpr(expr ast.ExprNode, ctx context.Context) (types.Datum, error) {
+	b := &planBuilder{
+		ctx:       ctx,
+		allocator: new(idAllocator),
+		colMapper: make(map[*ast.ColumnNameExpr]int),
+		is:        GetInfoSchema(ctx),
```
Comment (zimulala) [inline plan/expression_rewriter.go]: You can use `is: ctx.GetSessionVars().TxnCtx.InfoSchema.(infoschema.InfoSchema)` directly.

Comment (shenli) [inline plan/expression_rewriter.go]: +1

File: plan/expression_rewriter.go [hunk 2]
```diff
@@ -20,6 +20,24 @@ import (
 // EvalSubquery evaluates incorrelated subqueries once.
 var EvalSubquery func(p PhysicalPlan, is infoschema.InfoSchema, ctx context.Context) ([]types.Datum, error)
 
+// GetInfoSchema gets infoschema from ctx.
+var GetInfoSchema func(ctx context.Context) infoschema.InfoSchema
+
+// EvalAstExpr evaluates ast expression directly.
+func EvalAstExpr(expr ast.ExprNode, ctx context.Context) (types.Datum, error) {
+	b := &planBuilder{
```
Comment (shenli) [inline plan/expression_rewriter.go]: So we need to create a planBuilder for every expr?

Comment (hanfei1991) [inline plan/expression_rewriter.go]: Unless you can ensure the expression don't contains a subquery

File: plan/expression_rewriter.go [hunk 3]
```diff
@@ -20,6 +20,21 @@ import (
 // EvalSubquery evaluates incorrelated subqueries once.
 var EvalSubquery func(p PhysicalPlan, is infoschema.InfoSchema, ctx context.Context) ([]types.Datum, error)
 
+// EvalAstExpr evaluates ast expression directly.
+func EvalAstExpr(expr ast.ExprNode, ctx context.Context) (types.Datum, error) {
```
Comment (coocood) [inline plan/expression_rewriter.go]: check if expr is `ast.ValueExpr`, then we don't need to create planBuilder

**Comment by coocood** [comment]
LGTM

**Comment by hanfei1991** [comment]
@shenli @zimulala PTAL

**Comment by shenli** [comment]
LGTM

**Pull request merged**